packages/node/src/integrations/localvariables/localvariables-async.ts

```diff
@@ -0,0 +1,287 @@
+import type { Event, EventProcessor, Exception, Hub, Integration, StackFrame, StackParser } from '@sentry/types';
+import { LRUMap, logger } from '@sentry/utils';
+import type { Debugger, InspectorNotification, Runtime } from 'inspector';
+import type { NodeClient } from '../../client';
+
+import type { NodeClientOptions } from '../../types';
+import type { FrameVariables, Options, PausedExceptionEvent, RateLimitIncrement, Variables } from './common';
+import { createRateLimiter, functionNamesMatch, hashFrames, hashFromStack } from './common';
+
+type Session = {
+  connect: () => void;
+  new (): Session;
+
+  post(method: 'Debugger.pause' | 'Debugger.resume' | 'Debugger.enable' | 'Debugger.disable'): Promise<void>;
+  post(method: 'Debugger.setPauseOnExceptions', params: Debugger.SetPauseOnExceptionsParameterType): Promise<void>;
+  post(
+    method: 'Runtime.getProperties',
+    params: Runtime.GetPropertiesParameterType,
+  ): Promise<Runtime.GetPropertiesReturnType>;
+
+  on(
+    event: 'Debugger.paused',
+    listener: (message: InspectorNotification<Debugger.PausedEventDataType>) => void,
+  ): Session;
+
+  on(event: 'Debugger.resumed', listener: () => void): Session;
+};
+
+async function unrollArray(session: Session, objectId: string, name: string): Promise<Variables> {
+  const properties = await session.post('Runtime.getProperties', { objectId, ownProperties: true });
+
+  return {
+    [name]: properties.result
+      .filter(v => v.name !== 'length' && !isNaN(parseInt(v.name, 10)))
```

Comment:
`+val` is faster than `Number(val)` and `Number(val` is faster than `parseInt(val)`

```suggestion
      .filter(v => v.name !== 'length' && !Number.isNaN(+v.name))
```

Reply:
`+""` returns `0` whereas `parseInt("")` returns `NaN` so I'm not sure unary is a good choice here